From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Axel Garcia K." <git@axelgk.com>
Date: Tue, 5 May 2026 18:57:44 +0000
Subject: [PATCH 1/4] Avoid host CD-ROM I/O on SF2000

Use the cached-read path when platform CD-ROM command helpers are unavailable so the core does not try to call desktop host device APIs on HCRTOS.
---
 libretro-common/cdrom/cdrom.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libretro-common/cdrom/cdrom.c b/libretro-common/cdrom/cdrom.c
index 58cb5cf188dd78aeb4137c38d643adccb36cc90f..52d782aabe5d0e3e1b994e7d0f5b9325c0f954d7 100644
--- a/libretro-common/cdrom/cdrom.c
+++ b/libretro-common/cdrom/cdrom.c
@@ -486,6 +486,8 @@ retry:
 #else
 #if defined(_WIN32) && !defined(_XBOX)
       if (cached_read || !cdrom_send_command_win32(stream, dir, xfer_buf_pos, request_len, cmd, cmd_len, sense, sizeof(sense)))
+#else
+      if (cached_read)
 #endif
 #endif
       {

base-commit: 9ba79648d6ec85e833aef719d7f359117498d89c
-- 
2.54.0

